compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt

```diff
@@ -345,68 +345,73 @@ class RawFirBuilder(
+                        val outerContractDescription = this@toFirPropertyAccessor.obtainContractDescription()
+                        val bodyWithContractDescription = this@toFirPropertyAccessor.buildFirBody()
+                        this.body = bodyWithContractDescription.first
+                        val contractDescription = outerContractDescription ?: bodyWithContractDescription.second
+                        contractDescription?.let {
+                            this.contractDescription = it
+                        }
+                    }.also {
                         currentDispatchReceiverType()?.lookupTag?.let { lookupTag ->
                             it.containingClassAttr = lookupTag
                         }
+                        accessorTarget.bind(it)
+                        this@RawFirBuilder.context.firFunctionTargets.removeLast()
                     }
-            }
-            val source = this.toFirSourceElement()
-            val accessorTarget = FirFunctionTarget(labelName = null, isLambda = false)
-            return buildPropertyAccessor {
-                this.source = source
-                session = baseSession
-                origin = FirDeclarationOrigin.Source
-                returnTypeRef = if (isGetter) {
-                    returnTypeReference?.convertSafe() ?: propertyTypeRef
-                } else {
-                    returnTypeReference.toFirOrUnitType()
                 }
-                this.isGetter = isGetter
-                this.status = status
-                extractAnnotationsTo(this)
-                this@RawFirBuilder.context.firFunctionTargets += accessorTarget
-                extractValueParametersTo(this, propertyTypeRef)
-                if (!isGetter && valueParameters.isEmpty()) {
-                    valueParameters += buildDefaultSetterValueParameter {
-                        this.source = source.fakeElement(FirFakeSourceElementKind.DefaultAccessor)
-                        session = baseSession
-                        origin = FirDeclarationOrigin.Source
-                        returnTypeRef = propertyTypeRef
-                        symbol = FirVariableSymbol(NAME_FOR_DEFAULT_VALUE_PARAMETER)
-                    }
-                }
-                symbol = FirPropertyAccessorSymbol()
-                val outerContractDescription = this@toFirPropertyAccessor.obtainContractDescription()
-                val bodyWithContractDescription = this@toFirPropertyAccessor.buildFirBody()
-                this.body = bodyWithContractDescription.first
-                val contractDescription = outerContractDescription ?: bodyWithContractDescription.second
-                contractDescription?.let {
-                    this.contractDescription = it
-                }
-            }.also {
-                currentDispatchReceiverType()?.lookupTag?.let { lookupTag ->
-                    it.containingClassAttr = lookupTag
+                isGetter || property.isVar -> {
```

Comment:
IIUC there is no logic change in this file. Is this change just to improve the readability?

Reply:
This is really subtle, and thus need some code comments (as I described below).

Previously, if the receiver of `KtPropertyAccessor` type is not given or it doesn't have a body, we created a default accessor. For the remaining case---`KtPropertyAccessor` is given _and_ the body is also declared, we created a property accessor.

Now, with this change, we did the latter first, which is a clear case I think. Then, this check (`isGetter || property.isVar`) prevent us from creating a default accessor if it is setter inside `val` property. But, you said,
> The changes to FIR building were necessary as no `FirPropertyAccessor` was created for a setter inside a `val` property.

Thus, this change looks opposite; or the condition is somewhat incorrect; or maybe as @tgeng mentioned, you just tried to clean up? I think line 1277 below is the only necessary change?

Reply:
@jsjeon is correct, the only thing I added is to add `isGetter || property.isVar` when creating a default accessor, else return `null` (no accessor for setter in `val` property). And the rest is cosmetic, rearranging so the conditions are easier to read and smartcasting to not-null in the first branch works.

I will add some comments as suggested.

Reply:
With comments added, now things look much clearer! 👍

Reply:
Just to make sure I understand it correctly. So the changes in this file are indeed cosmetic and introduce no behavior change, right?

Reply:
Sorry it is not clear. There _is_ a behavior change from the added condition and the `else null` which is new (previously the function returns not-nullable), as explained in my previous post.

Reply:
Yeah, but the caller is changed too. And it seems to me that all the changes in this file combined cause no behavior changes.